members/models/union.py

```diff
@@ -1,54 +1,115 @@
-            raise ValidationError(
-                "Vælg om foreningen har konto hos hovedforeningen. Hvis ikke skal bankkonto udfyldes."
-            )
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from django.db import models
+import members.models.payment as Payment
+import members.models.activity as Activity
+import members.models.activityparticipant as ActivityParticipant
+from django.core.exceptions import ValidationError
+from django.core.validators import RegexValidator
+from django.utils import timezone
+from datetime import timedelta
+from django.db.models import F
+import datetime
+
+
+class Union(models.Model):
+    class Meta:
+        verbose_name_plural = "Foreninger"
+        verbose_name = "Forening"
+        ordering = ["name"]
+
+    name = models.CharField("Foreningens navn", max_length=200)
+    chairman = models.CharField("Formand", max_length=200, blank=True)
+    chairman_email = models.EmailField("Formandens email", blank=True)
+    second_chair = models.CharField("Næstformand", max_length=200, blank=True)
+    second_chair_email = models.EmailField("Næstformandens email", blank=True)
+    cashier = models.CharField("Kasserer", max_length=200, blank=True)
+    cashier_email = models.EmailField("Kassererens email", blank=True)
+    secretary = models.CharField("Sekretær", max_length=200, blank=True)
+    secratary_email = models.EmailField("Sekretærens email", blank=True)
+    union_email = models.EmailField("Foreningens email", blank=True)
+    statues = models.URLField("Link til gældende vedtægter", blank=True)
+    meeting_notes = models.URLField("Link til seneste referater", blank=True)
+    founded = models.DateField("Stiftet", blank=True, null=True)
+    regions = (("S", "Sjælland"), ("J", "Jylland"), ("F", "Fyn"), ("Ø", "Øer"))
+    region = models.CharField("region", max_length=1, choices=regions)
+    address = models.ForeignKey("Address", on_delete=models.PROTECT)
+    boardMembers = models.TextField("Menige medlemmer", blank=True)
+    bank_main_org = models.BooleanField(
+        "Sæt kryds hvis I har konto hos hovedforeningen (og ikke har egen bankkonto).",
+        default=True,
+    )
+    bank_account = models.CharField(
+        "Bankkonto:",
+        max_length=15,
+        blank=True,
+        help_text="Kontonummer i formatet 1234-1234567890",
+        validators=[
+            RegexValidator(
+                regex="^[0-9]{4} *?-? *?[0-9]{6,10} *?$",
+                message="Indtast kontonummer i det rigtige format.",
+            )
+        ],
+    )
+
+    def __str__(self):
+        return "Foreningen for " + self.name
+
+    def clean(self):
+        if self.bank_main_org is False and not self.bank_account:
+            raise ValidationError(
+                "Vælg om foreningen har konto hos hovedforeningen. Hvis ikke skal bankkonto udfyldes."
+            )
+
+    def members(self):
+        years = range(self.founded.year, (timezone.now().date()).year + 1)
```

Comment:
Den her linje fejler hvis `founded` ikke er sat. 
Det er der fire på prod der ikke har. 

Mit forslag er at vi manuelt går ind og sætter dem på prod så de alle passer. 
Så gør vi `founded` required og sætter en default værdi i migration.

Reply:
Ok, skal jeg bare sætte den til reguired, default 2014 eller sådan noget, nu?

Reply:
Sæt den til required, men lad være med at sæt default på modellen. 
I stedet skal du sætte default i den migration det kommer til at give. 
Når du kører `makemigrations` så spørger den om en default, der kan du give en 
værdi som 1970 eller noget så vi tydeligt kan se at det er "fake" data og rette det. 

Fordellen ved at gøre det i en migration i stedet for på modellen er at den så kun gælder bagudrettet og nye oprettede foreninger ikke defaulter til den dato. 
![Screenshot 2020-04-15 at 11 44 12](https://user-images.githubusercontent.com/3578071/79323398-735d8580-7f0e-11ea-8f70-81ac70587ab0.png)

Det er kun de fire foreninger der ikke har en dato. Så kan du evt. spørge Michael om datoer om og så rette manuelt. Selvom du retter dem manuelt kræver det stadig en default i migration, den bliver så bare ikke brugt på prod. Men bliver brugt på staging, review apps og vores lokale setups.

Reply:
Ok, done